instrumentation/spring/spring-boot-resources/library/src/main/java/io/opentelemetry/instrumentation/spring/resources/SpringBootServiceNameDetector.java

```diff
@@ -45,6 +45,9 @@
  *   <li>Check for --spring.application.name program argument (not jvm arg) via ProcessHandle
  *   <li>Check for --spring.application.name program argument via sun.java.command system property
  * </ul>
+ *
+ * <p>Note: should not be used inside a spring application, where the spring.application.name is
+ * already available.
```

Comment:
I didn't understand this note?

Reply:
sorry, still not following

Reply:
the spring boot starter uses [SpringResourceProvider](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/d8aa0f5b48e2dee28498b6f95d3d0017eb673cbf/instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/resources/SpringResourceProvider.java), which doesn't need to parse any application.yaml, etc. file, because when you run inside a sprig application, you can just ask the environment.

In this case, the lookup is here: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/d8aa0f5b48e2dee28498b6f95d3d0017eb673cbf/instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/resources/SpringConfigProperties.java#L45

Reply:
yes, exactly.

The provider will actually notice if a name found using `shouldApply` - so it's just a note about the intended usage.

Reply:
I've done this move in https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/10621 - because this 

PR moves `SystemHelper` to the common resource project (for other reasons). 
Hope this makes sense.